testing/scenario_app/android/README.md

```diff
@@ -1,102 +1,364 @@
+
+```sh
+export ANDROID_HOME=$ENGINE/src/third_party/android_tools/sdk
+```
+
+### Device or Emulator
+
+The tests in this package require a connected device or emulator to run. The
+device or emulator should be running the same version of Android as the CI
+configuration (there are issues with crashes and other problems on older
+emulators in particular).
+
+> [!CAUTION]
+>
+> [#144561](https://github.com/flutter/flutter/issues/144561): The emulator
+> vendored in the engine checkout is old and [has a known issue with Vulkan](https://github.com/flutter/flutter/issues/144232).
+>
+> If you're working locally, you can update your copy by running:
+>
+> ```sh
+> $ANDROID_HOME/cmdline-tools/latest/bin/sdkmanager --install emulator
+> ```
+
+### Additional Dependencies
+
+While not required, it is **strongly recommended** to use an IDE such as
+[Android Studio](https://developer.android.com/studio) when contributing to the
+Android side of the test suite, as it will provide a better development
+experience such as code completion, debugging, and more:
+
+<img
+  width="500"
+  alt="Screenshot of Android Studio with Autocompletion"
+  src="https://github.com/flutter/flutter/assets/168174/79b685f5-8da7-4396-abe6-9baeb29d7ce3"
+/>
+
+> [!TIP]
+>
+> Android Studio is expected to work _out of the box_ in this directory.
+>
+> If you encounter any issues, please [file a bug](#getting-help).
+
+## Running the Tests
+
+The [test runner](../bin/run_android_tests.dart) is a Dart script that installs
+the test app and test suite on a connected device or emulator, runs the tests,
+and reports the results including screenshots (golden files) and logs from
+`adb logcat`.
+
+From the `$ENGINE/src/flutter` directory, run:
 
 ```sh
 dart ./testing/scenario_app/bin/run_android_tests.dart
 ```
 
-Or for a specific, build, such as `android_debug_unopt_arm64`:
+By default when run locally, the test runner:
+
+- Uses the engine-wide efault graphics backend for Android.
+- Uses the last built Android-specific engine artifacts (i.e. `$ENGINE/src/out/android_*/`).
+- Will not diff screenshots, but does save them to a logs directory.
+
+### Rebuilding the Tests
+
+If you've made changes to any file in `scenario_app`, incluing the Dart code
+in `lib/` or the Android code in `android/`, you will need to rebuild the
+tests before running them.
+
+To rebuild the `scenario_app` for the `android_debug_unopt_arm64` variant:
 
 ```sh
-dart ./testing/scenario_app/bin/run_android_tests.dart --out-dir=../out/android_debug_unopt_arm64
+ninja -C out/android_debug_unopt_arm64 scenario_app
 ```
 
 See also:
 
-- [File an issue][file_issue] with the `e: scenario-app, platform-android`
-  labels.
+- [Compiling the Engine](https://github.com/flutter/flutter/wiki/Compiling-the-engine)
+
+### Common Options
+
+A list of options can be found by running:
+
+```sh
+dart ./testing/scenario_app/bin/run_android_tests.dart --help
+```
+
+Frequency used options include:
```

Comment:
```suggestion
Frequently used options include:
```

Reply:
fyi: there is a vscode plugin that will do spell checking, super handy.